### PR TITLE
nixos-shell:init at  2015-05-23

### PR DESCRIPTION
**Pull request by pasqui23**
###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by teto** [comment]
isn't the upstream at https://github.com/Mic92/nixos-shell ?

**Comment by pasqui23** [comment]
You're right,and it's even more recently updated

**Comment by pasqui23** [comment]
That one has the same name but it's different:it spawns a whole new virtual machine instead of just a container

**Comment by tomberek** [comment]
nixos-container is required at runtime, but is not in the dependencies nor propagated. And should it be limited to NixOS? or can someone run nixos-container on non-NixOS?

**Comment by pasqui23** [comment]
@tomberek it is merely a thin wrapper aroun nixos-container,so it should run where nixos-container runs

**Comment by pasqui23** [comment]
Ok,why the check meta fails?

File: pkgs/tools/package-management/nixos-shell/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  buildPhase =
+    ''
+    mkdir -p $out/bin
+    GOPATH=$src HOME="." go build -o $out/bin/nixos-shell nixos-shell
+    '';
+  meta = with stdenv.lib;{
+    homepage = "https://github.com/wavewave/nixos-shell" ;
+    description = "Spawn ephemeral development environments in containers";
+    license = licenses.free;
+    platform = platforms.linux;
```
Comment (c0bw3b) [inline pkgs/tools/package-management/nixos-shell/default.nix]: It should be
```suggestion
    platforms = platforms.linux;
```

**Comment by doronbehar** [comment]
Closed in favor of https://github.com/NixOS/nixpkgs/pull/85673 .

**Pull request closed without merging**